doc/cli.md

```diff
@@ -0,0 +1,230 @@
+  --template-id https://xapinet.org/xapi/yet/calibration/v1/templates#activity-2
+  --template-id https://xapinet.org/xapi/yet/calibration/v1/templates#activity-3
+```
+This way, we pass to the CLI the two Templates the Statement is invalid against, so running this will result in the same validation failure output as in the previous example.
+
+Note that if we set `--template-id` to the ID of a non-existent Template, e.g.
+```
+% ./bin/persephone.sh validate \
+  --profile sample_profiles/calibration.jsonld \
+  --statement sample_statements/calibration_1.json \
+  --template-id http://random-template.org
+```
+the validation will pass vacuously, as there are technically no Templates the Statement is invalid against.
+
+## Examples for `persephone match`
+
+We can match the Statement batch consisting of `calibration_1.json` and `calibration_2.json` against the Patterns in the `calibration.jsonld` Profile as so:
+```
+% ./bin/persephone.sh match \
+  --profile sample_profiles/calibration.jsonld \
+  --statement sample_statements/calibration_1.json \
+  --statement sample_statements/calibration_2.json
+```
+This will run, print nothing, and exit with code 0, indicating match success.
+
+However, if we were to match a Statement that was not intended to be matched with the Profile, for example:
+```
+% ./bin/persephone.sh match \
+  --profile sample_profiles/calibration.jsonld \
+  --statement test-resources/sample_statements/adl_3.json
+```
+the CLI will print an error message to stdout indicating that the Statement does not refer to the profile version in its category context activity IDs:
+```
+----- Pattern Match Error -----
+Error Description:  Missing Profile version in context category activity IDs
+Statement ID:       6690e6c9-3ef0-4ed3-8b37-7f3964730bee
+
+Category contextActivity IDs:
+http://www.example.com/meetings/categories/teammeeting
+```
+
+Errors would also appear if the Statement included a sub-registration without a registration, or if the sub-registration did not conform to spec.
+
+The order in which the Statements are passed to the `match` subcommand matters greatly. If we switch the order of our Statements from the first example, Pattern matching fails:
+```
+% ./bin/persephone.sh match \
+  --profile sample_profiles/calibration.jsonld \
+  --statement sample_statements/calibration_2.json \
+  --statement sample_statements/calibration_1.json
+```
+and this message will be printed to stdout:
+```
+----- Pattern Match Failure -----
+Primary Pattern ID: https://xapinet.org/xapi/yet/calibration/v1/patterns#pattern-1
+Statement ID:       00000000-4000-8000-0000-000000000000
+
+Pattern matching has failed.
+```
+
+If we set the `--compile-nfa` flag, then we can get a detailed trace of the Pattern matching path:
+```
+% ./bin/persephone.sh match \
+  --profile sample_profiles/calibration.jsonld \
+  --statement sample_statements/calibration_2.json \
+  --statement sample_statements/calibration_1.json \
+  --compile-nfa
```

Comment:
`--compile-nfa` only seems to work when matching fails. Is this intentional? I could absolutely see the pathing output being useful whether or not the matching fails or succeeds.

Reply:
Yes that is intentional. The initial matching is done on a DFA regardless of whether the flag is set or not, for efficiency reasons. The NFA only comes in when there's a match failure and you need to retrace your steps.

Reply:
Hmm, but without this output how would you tell a partial match from a complete one?

Reply:
<div class="tenor-gif-embed" data-postid="27194608" data-share-method="host" data-aspect-ratio="1.77778" data-width="100%"><a href="https://tenor.com/view/thats-the-neat-part-you-dont-invincible-gif-27194608">Thats The Neat Part You Dont Invincible GIF</a>from <a href="https://tenor.com/search/thats+the+neat+part+you+dont-gifs">Thats The Neat Part You Dont GIFs</a></div> <script type="text/javascript" async src="https://tenor.com/embed.js"></script>

Reply:
In all seriousness, you would tell if a statement batch was fully accepted by checking the `:accepted` value in the result map. But even then you might be able to do further matching so the line between "partial" and "complete" matches get blurry.

Reply:
Right, but is the result map something you get back in the CLI?

Reply:
No...I thought about printing that, but it gets troublesome because if you print it in `match-statement` and keep calling the function (e.g. as part of a stream) you'll get lots of stdout spam.

Reply:
Maybe it's not something we need (describing the matched path on a success, and thus being able to tell partial matches from full ones), I leave it to @cliffcaseyyet.

Reply:
Per @cliffcaseyyet on slack we don't need to differentiate partial and full matches.